coverage-tool/coverage-tool.py

```diff
@@ -0,0 +1,300 @@
+            files.update(matches)
+        return files
+
+    parser = ArgumentParser()
+    parser.add_argument(
+        "--percent-only",
+        "-p",
+        action="store_true",
+        help="Print only the total coverage percentage",
+    )
+    parser.add_argument(
+        "--json-output",
+        "-j",
+        action="store_true",
+        help="Output the coverage data as JSON",
+    )
+    parser.add_argument(
+        "--list-files",
+        "-l",
+        action="store_true",
+        help="Output list of mock response files that cover each field",
+    )
+    parser.add_argument(
+        "--no-color",
+        "-n",
+        action="store_true",
+        help="Disable color in output",
+    )
+    parser.add_argument(
+        "--ignore-fields",
+        "-i",
+        metavar="FIELDS",
+        type=str,
+        help="Ignore the fields in the given comma-separated list when calculating"
+        " coverage",
+    )
+    parser.add_argument(
+        "--scan-files",
+        "-s",
+        metavar="PATTERN",
+        type=str,
+        nargs="+",
+        help="Scan files matching the given pattern(s) for names of mock response files"
+        " (without the extension) to evaluate",
+    )
+    parser.add_argument(
+        "--exclude",
+        "-e",
+        metavar="PATTERN",
+        type=str,
+        nargs="+",
+        help="Exclude mock response files matching the given pattern(s) from being"
+        " evaluated",
+    )
+    args = parser.parse_args()
+    if args.percent_only and (args.json_output or args.list_files):
+        parser.error(
+            "`--percent-only` cannot be used with `--json-output` or `--list-files`."
+        )
+    args.ignore_fields = (
+        set(args.ignore_fields.split(",")) if args.ignore_fields else set()
+    )
+    args.scan_files = get_files_from_patterns(args.scan_files)
+    args.exclude = get_files_from_patterns(args.exclude, MOCK_RESPONSES_PATH)
+    return args
+
+
+def get_schemas():
+    """Download discovery document and return its schemas."""
+    doc_path = os.path.join(SCRIPT_DIR, "discovery_doc.json")
+    if not os.path.exists(doc_path):
+        with open(doc_path, "w") as doc:
+            with urlopen(DISCOVERY_DOC_URL) as response:
+                doc.write(response.read().decode("utf-8"))
+    return json.load(open(doc_path))["schemas"]
+
+
+def get_mock_responses_list():
+    """Return a list of mock response files to evaluate according to args."""
+    mock_responses = sorted(os.listdir(MOCK_RESPONSES_PATH))
+    if args.exclude:
+        mock_responses = [f for f in mock_responses if f not in args.exclude]
+    if args.scan_files:
+        content = ""
+        for file_path in args.scan_files:
+            with open(file_path) as f:
+                content += f.read()
+        mock_responses = [f for f in mock_responses if f.split(".")[0] in content]
+    return mock_responses
+
+
+def load_mock_response(file_path):
```

Comment:
The way you have to load streaming, and non streaming data is different enough that different methods would make it easier to understand how they work.

Reply:
Problem: No data will be extracted from lines that start with anything other than `data:` in streaming files.
Solution: We print a warning if nothing was extracted from a file.

Reply:
was this addressed?

Reply:
yes, I added [the check](https://github.com/FirebaseExtended/vertexai-sdk-test-data/pull/8/files#diff-e4df3c0f0626af8f9a28ea8d16c2d7f0c11e4f2c8bf9c4bfc7d01dd9aa02f720R127) to `get_grouped_mock_responses()` to be able to exclude that file and to keep both warning prints in the same function